spyder/plugins/maininterpreter.py

```diff
@@ -222,5 +230,20 @@ def set_umr_namelist(self):
                 fixed_namelist = []
             self.set_option('umr/namelist', fixed_namelist)
 
+    def set_custom_interpreters_list(self, executable):
+        """Update the list of interpreters used and the current one."""
+        custom_list = self.get_option('custom_list')
+        if (executable, executable) not in custom_list:

```

Comment:
This looks confusing, i.e. naming two variables with the same name (`execuatble`, `executable`). Please change that.